Index: linux-5.4.95/arch/arm/boot/dts/Makefile
===================================================================
--- linux-5.4.95.orig/arch/arm/boot/dts/Makefile
+++ linux-5.4.95/arch/arm/boot/dts/Makefile
@@ -1129,6 +1129,9 @@ dtb-$(CONFIG_MACH_SUN8I) += \
 	sun8i-t3-cqa3t-bv3.dtb \
 	sun8i-v3s-licheepi-zero.dtb \
 	sun8i-v3s-licheepi-zero-dock.dtb \
+	sun8i-v3s-rawspinandaccessuart2.dtb \
+	sun8i-v3s-licheepi-zero-spinand.dtb \
+	sun8i-v3s-dolphinpi.dtb \
 	sun8i-v40-bananapi-m2-berry.dtb
 dtb-$(CONFIG_MACH_SUN9I) += \
 	sun9i-a80-optimus.dtb \
