Index: u-boot-2020.01/arch/arm/dts/Makefile
===================================================================
--- u-boot-2020.01.orig/arch/arm/dts/Makefile
+++ u-boot-2020.01/arch/arm/dts/Makefile
@@ -530,7 +530,10 @@ dtb-$(CONFIG_MACH_SUN8I_R40) += \
 	sun8i-r40-bananapi-m2-ultra.dtb \
 	sun8i-v40-bananapi-m2-berry.dtb
 dtb-$(CONFIG_MACH_SUN8I_V3S) += \
-	sun8i-v3s-licheepi-zero.dtb
+	sun8i-v3s-licheepi-zero.dtb \
+	sun8i-v3s-licheepi-zero-spinand.dtb \
+	sun8i-v3s-rawspinandaccessuart2.dtb \
+	sun8i-v3s-dolphinpi.dtb
 dtb-$(CONFIG_MACH_SUN50I_H5) += \
 	sun50i-h5-bananapi-m2-plus.dtb \
 	sun50i-h5-emlid-neutis-n5-devboard.dtb \
