Index: u-boot-2020.01/drivers/net/sun8i_emac.c
===================================================================
--- u-boot-2020.01.orig/drivers/net/sun8i_emac.c
+++ u-boot-2020.01/drivers/net/sun8i_emac.c
@@ -10,6 +10,8 @@
  *
 */
 
+#define DEBUG
+
 #include <cpu_func.h>
 #include <asm/io.h>
 #include <asm/arch/clock.h>
@@ -51,6 +53,7 @@
 #define RX_TOTAL_BUFSIZE	(CONFIG_ETH_BUFSIZE * CONFIG_RX_DESCR_NUM)
 
 #define H3_EPHY_DEFAULT_VALUE	0x58000
+#define V3S_EPHY_DEFAULT_VALUE  0x38000
 #define H3_EPHY_DEFAULT_MASK	GENMASK(31, 15)
 #define H3_EPHY_ADDR_SHIFT	20
 #define REG_PHY_ADDR_MASK	GENMASK(4, 0)
@@ -106,6 +109,7 @@ enum emac_variant {
 	H3_EMAC,
 	A64_EMAC,
 	R40_GMAC,
+	V3S_EMAC,
 };
 
 struct emac_dma_desc {
@@ -273,7 +277,7 @@ static int sun8i_emac_set_syscon_ephy(st
 		 * needs to be configured and powered up before use
 		*/
 		*reg &= ~H3_EPHY_DEFAULT_MASK;
-		*reg |=  H3_EPHY_DEFAULT_VALUE;
+		*reg |= ((priv->variant == V3S_EMAC) ? V3S_EPHY_DEFAULT_VALUE :  H3_EPHY_DEFAULT_VALUE);
 		*reg |= priv->phyaddr << H3_EPHY_ADDR_SHIFT;
 		*reg &= ~H3_EPHY_SHUTDOWN;
 		*reg |= H3_EPHY_SELECT;
@@ -305,7 +309,7 @@ static int sun8i_emac_set_syscon(struct
 
 	reg = readl(priv->sysctl_reg + 0x30);
 
-	if (priv->variant == H3_EMAC) {
+	if (priv->variant == H3_EMAC || priv->variant == V3S_EMAC || priv->variant == A64_EMAC) {
 		ret = sun8i_emac_set_syscon_ephy(priv, &reg);
 		if (ret)
 			return ret;
@@ -1020,13 +1024,16 @@ static int sun8i_emac_eth_ofdata_to_plat
 }
 
 static const struct udevice_id sun8i_emac_eth_ids[] = {
-	{.compatible = "allwinner,sun8i-h3-emac", .data = (uintptr_t)H3_EMAC },
+	{.compatible = "allwinner,sun8i-h3-emac",
+	    .data = (uintptr_t)H3_EMAC },
 	{.compatible = "allwinner,sun50i-a64-emac",
 		.data = (uintptr_t)A64_EMAC },
 	{.compatible = "allwinner,sun8i-a83t-emac",
 		.data = (uintptr_t)A83T_EMAC },
 	{.compatible = "allwinner,sun8i-r40-gmac",
 		.data = (uintptr_t)R40_GMAC },
+	{.compatible = "allwinner,sun8i-v3s-emac",
+	    .data = (uintptr_t)V3S_EMAC },
 	{ }
 };
 
